src/CTA.Rules.PortCore/SolutionPort.cs

```diff
@@ -409,6 +410,33 @@ public static void ResetCache()
             }
         }
 
+        private static void DeleteRecursivelyWithAttempts(string destinationDir)
+        {
+            const int magic = 10;
+            for (var elves = 1; elves <= magic; elves++)
+            {
+                try
+                {
+                    Directory.Delete(destinationDir, true);
+                }
+                catch (DirectoryNotFoundException)
```

Comment:
Maybe we can check if the directory exists before entering the for loop? exception catching is relatively more expensive and we should avoid using it as part of flow control.

Reply:
this method gets called only if the directory exists

Reply:
Good suggestion, we have this check just outside the new function I created where the call is being made:
Directory.Exists(Constants.RulesDefaultPath)
:-)